### PR TITLE
Fix build on Plan 9

### PR DESCRIPTION
**Pull request by fhs**
All test are passing on Plan 9, and I've also verified
`examples/go-sftp-server` is working.

### CODE REVIEW

File: attrs.go
Comment (puellanivis) [inline attrs.go:195]: Huh… one would think that these could be set as (for example) `syscall.S_ISGID = 0` and then this would be recognized as always false, and thus optimized out for us…

I‘m sure there are subtle reasons why they chose to do so.

File: stat_unix.go
Comment (puellanivis) [inline stat_unix.go:1]: I would expect a `_unix.go` file would also exclude `!windows`, and likely `!js` as well. But then they are compatible with this code only because Go provides “fictitious” values so that they are otherwise compable without code changes like we‘ve had to do here for plan9…

Not sure I have an answer for what to do here, though. I think technically, it‘s `_posix.go`. But only because it‘s the part of POSIX that we‘re able to build support for out of the very rudimentary baseline WIndows POSIX support… Maybe `_posix_compat.go`? or just `_compat.go`?

File: server.go
Comment (puellanivis) [inline server.go:578]: From the other PR, for better Windows compat of a file underneath a non-existing directory, we probably want to switch to using `if os.IsNotExist(err) { ret.StatusError.Code = sshFxNoSuchFile ; return ret }` which should simplify up a lot of your code handling here? As after, we would not need any special syscall error code handling for file-not-found errors.

**Review by puellanivis — COMMENTED** [review]
Probably pretty good. Really more nits/lint than anything?

File: attrs.go
Comment (fhs) [reply attrs.go:195]: The `syscall` package is not meant to be portable. It's actually currently frozen, so I think don't we can add `S_ISGID` for plan9 even if we wanted to. In fact, the `syscall` package might be [removed](https://github.com/golang/go/issues/5616#issuecomment-349783401) in the furture in favor of `golang.org/x/sys`.

Comment (puellanivis) [reply attrs.go:195]: Indeed. And that‘s probably a good thing. But it‘s weird that they stopped at extended compatibility for windows and JS, but not for plan9.

Maybe it lost priority, and didn‘t keep up? Not sure. Just seems weird in my head to make all the fictional values for windows and JS, but not for plan9. :laughing:

File: stat_unix.go
Comment (fhs) [reply stat_unix.go:1]: I'll use `_posix.go` because that's what the standard library uses.

File: server.go
Comment (fhs) [reply server.go:578]: Yeah, it's better to use `os.IsNotExist` and `os.Permission` here. I was trying to keep the changes to minimum, but I can certainly make that change.

Comment (fhs) [reply server.go:578]: We still need syscall error code handing for the case when `errno == 0` (we return `sshFxOk`), and when errno is non-zero but not `ENOENT` or `EPERM`, in which case we return `sshFxFailure`.

I'm inclined to leave this change out of this PR, since we're just trying to fix build on plan9 (and not break other systems in the process).

File: stat_unix.go
Comment (fhs) [reply stat_unix.go]: Done

File: server.go
Comment (puellanivis) [reply server.go:578]: I’m going to recommend that you hold off of that change. We’ll probably merge the other PR before yours, and then you’ll have a merge conflict to clean up, but hopefully it shouldn’t be too bad. Especially, since their code only touches a strict subset of your code, so you should be pretty good and familiar with the code to make the right conflict resolution.

Comment (puellanivis) [reply server.go:578]: The other PR has been merged, and as expected, github now reports there are merge conflicts to resolve. :smiley: 

Once you‘ve addressed those, I think we‘re good to commit. Thanks for the work! I know it can be hard to find people who can even test on plan9, let alone do any support work necessary.

Comment (fhs) [reply server.go:582]: Done. Thanks for the review!

File: stat_posix.go
Comment (puellanivis) [inline stat_posix.go:13]: Now that I look at this, this could probably better be:
```
if errno, ok := err.(syscall.Errno); ok {
  return &os.PathError{Path: filepath, Err: errno }
}
return err
```

With something similar for the plan9 with `ErrorString` instead of `Errno`.

A good compiler would make both equivalent, so this really should be about what is more readable, and/or if we expect this type-switch to be further expanded.

Your input is desired regarding this matter. It’s probably fine either way.

Comment (fhs) [reply stat_posix.go]: I agree that an if-statement is better here. It was using a switch in the original code, so I just left it like that.

I've converted it to an if-statement.

**Review by puellanivis — APPROVED** [review]

**Comment by puellanivis** [comment]
I don‘t see anything. I‘ll leave open a day for anyone else who might want to chime in.

**Pull request merged**
